src/coord/hybrid/dist_hybrid_plan.cpp

```diff
@@ -95,17 +95,32 @@ arrayof(char*) HybridRequest_BuildDistributedPipeline(HybridRequest *hreq,
     RLookup_Init(tailLookup, IndexSpec_GetSpecCache(hreq->sctx->spec));
 
     int rc = HybridRequest_BuildDistributedDepletionPipeline(hreq, hybridParams);
-    if (rc != REDISMODULE_OK) return NULL;
+    if (rc != REDISMODULE_OK) {
+      // The error is set at either the tail or the subquries error array
+      // need to copy it to the status so it will be visible to the user
+      HybridRequest_GetError(hreq, status);
+      HybridRequest_ClearErrors(hreq);
+      return NULL;
+    }
+
+    // Add keys from all source lookups to create unified schema before opening the score key
+    HybridRequest_SyncrhonizeLookupKeys(hreq);
```

Comment:
can we rename the function to fix the typo?
```suggestion
    HybridRequest_SynchronizeLookupKeys(hreq);
```